explorer/src/url_state.js

```diff
@@ -33,25 +33,32 @@ export function iniUrl() {
       initMap(State.center, State.zoom)
     }
   }
-
 }
 
-export function update(o) {
+export function updateState(o) {
   Object.assign(State, o)
   let urlhash = `#/${State.zoom}/${State.center[0]}/${State.center[1]}${State.hierarchyId ? `/${State.hierarchyId}` : ''}`
   if(history && typeof history.replaceState === 'undefined') {
     location.replace(urlhash)
   } else {
     history.replaceState(null, null, urlhash);
   }
+
+}
+export function pushState(o) {
+  Object.assign(State, o)
+  let urlhash = `#/${State.zoom}/${State.center[0]}/${State.center[1]}${State.hierarchyId ? `/${State.hierarchyId}` : ''}`
+  if(history && typeof history.pushState === 'undefined') {
+    location.replace(urlhash)
+  } else {
+    history.pushState(null, null, urlhash);
+  }
 }
 
 window.onpopstate = function () {
-  /* update only for hierarchy navigation */
   if(window.location.hash.split('/').length > 3) {
-    let tempHierarchyId = parseInt(window.location.hash.split('/')[3])
-    if(tempHierarchyId !== State.hierarchyId) {
-      fire('update_hierarchy', tempHierarchyId)
-    }
+    let tempHierarchyId = parseInt(window.location.hash.split('/')[4])
+    /* update only for hierarchy navigation */
+    fire('update_hierarchy', tempHierarchyId, {skipPushState : true})
```

Comment:
As far as I understand, this refreshes the panel on the left when browsing history. But the map zoom and position are not updated.
Also, when going back to a zone that was previously selected, the selection style (with red border) is not applied and stays applied on the last zone.